examples/md-flexible/src/Simulation.h

```diff
@@ -508,3 +522,69 @@ void Simulation<Particle, ParticleCell>::writeVTKFile(unsigned int iteration,
 
   _timers.vtk.stop();
 }
+
+template <class Particle, class ParticleCell>
+double Simulation<Particle, ParticleCell>::getHomogeneity(autopas::AutoPas<Particle, ParticleCell> &autopas) {
+  int numberOfParticles = autopas.getNumberOfParticles();
+  int numberOfCells = ceil(numberOfParticles / 10);
+
+  std::array<double, 3> startCorner = autopas.getBoxMin();
+  std::array<double, 3> endCorner = autopas.getBoxMax();
+  std::array<double, 3> domainSizePerDimension = {};
+  for (int i = 0; i < 3; ++i) {
+    domainSizePerDimension[i] = endCorner[i] - startCorner[i];
+  }
+
+  // get cellLength which is equal in each direction, derived from the domainsize and the requested number of cells
+  double volume = domainSizePerDimension[0] * domainSizePerDimension[1] * domainSizePerDimension[2];
+  double cellVolume = volume / numberOfCells;
+  double cellLength = cbrt(cellVolume);
+
+  // calculate the size of the boundary cells, which might be smaller then the other cells
+  std::array<long, 3> cellsPerDimension = {};
+  std::array<double, 3> outerCellSizePerDimension = {};
+  for (int i = 0; i < 3; ++i) {
+    outerCellSizePerDimension[i] =
+        domainSizePerDimension[i] - (floor(domainSizePerDimension[i] / cellLength) * cellLength);
+    cellsPerDimension[i] = ceil(domainSizePerDimension[i] / cellLength);
+  }
+  numberOfCells = cellsPerDimension[0] * cellsPerDimension[1] * cellsPerDimension[2];
+
+  std::vector<int> allCells(numberOfCells, 0);
+  std::vector<double> allVolumes(numberOfCells, 0);
+
+  // add particles accordingly to their cell to get the amount of particles in each cell
+  for (auto iter = autopas.begin(); iter.isValid(); ++iter) {
+    std::array<double, 3> region = iter->getR();
+    std::array<long, 3> index = {};
+    for (int i = 0; i < region.size(); i++) {
+      index[i] = region[i] / cellLength;
+    }
+    const unsigned long cellIndex = autopas::utils::ThreeDimensionalMapping::threeToOneD(index, cellsPerDimension);
+    allCells[cellIndex] += 1;
+    allVolumes[cellIndex] = (index[0] == cellsPerDimension[0] - 1) ? outerCellSizePerDimension[0] : cellLength;
+    allVolumes[cellIndex] *= (index[1] == cellsPerDimension[1] - 1) ? outerCellSizePerDimension[1] : cellLength;
+    allVolumes[cellIndex] *= (index[2] == cellsPerDimension[2] - 1) ? outerCellSizePerDimension[2] : cellLength;
```

Comment:
```suggestion
    // calculate the size of the current cell
    allVolumes[cellIndex] = 0;
    for (int i = 0; i < cellsPerDimension.size(); ++i) {
      // the last cell layer has a special size
      if (index[i] == cellsPerDimension[i] - 1) {
        allVolumes[cellIndex] *= outerCellSizePerDimension[i];
      } else {
        allVolumes[cellIndex] *=  cellLength;
      }
    }
```
Also use a for loop instead of repeating lines + ternary operator ;)